Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -500,46 +506,27 @@ void ReadTrackingState()
             {
                 // Treat an Input Action Reference with no reference the same as
                 // an enabled Input Action with no authored bindings, and allow driving the Transform pose.
-                m_CurrentTrackingState = TrackingStates.Position | TrackingStates.Rotation;
-                return;
+                var hasPositionInputActionWithBindings = HasResolvedControl(positionInput.action);
+                var hasRotationInputActionWithBindings = HasResolvedControl(rotationInput.action);
```

Comment:
For consistency, should we use the private field here as we did above?

Also, we could save on these 'expensive' `HasResolvedControl` calls by using the previous calculation in the `UpdateCallback` method and passing it into the `ReadTrackingState` as parameters.

```suggestion
                var hasPositionInputActionWithBindings = HasResolvedControl(m_PositionInput.action);
                var hasRotationInputActionWithBindings = HasResolvedControl(m_RotationInput.action);
```

Reply:
Sure, no strong opinion, will update and post commit ID once done.

Reply:
Suggestion to avoid calculation is great, overlooked it since it was split into two functions

Reply:
Addressed in https://github.com/Unity-Technologies/InputSystem/pull/1823/commits/6b5c6a2af9288169032ebf9b3364e8fd2de18ae7, feel free to resolve if sufficient.